### PR TITLE
new PR

### PR DESCRIPTION
**Pull request by layzimayner**

### CODE REVIEW

File: src/main/java/com/example/demo/config/AppConfig.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/config/AppConfig.java]: remove

File: src/main/java/com/example/demo/model/Book.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/model/Book.java]: don't use Data for entities

File: src/main/java/com/example/demo/model/Book.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/model/Book.java:17]: add `@Table`

File: src/main/java/com/example/demo/model/Book.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/model/Book.java:22]: add required constraints

File: src/main/java/com/example/demo/repository/BookRepositoryImpl.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/repository/BookRepositoryImpl.java:15]:
```suggestion
@Repository
@RequiredArgsConstructor
```

File: src/main/java/com/example/demo/repository/BookRepositoryImpl.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/repository/BookRepositoryImpl.java]:
```suggestion
    private final SessionFactory sessionFactory;
```

File: src/main/java/com/example/demo/repository/BookRepositoryImpl.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/repository/BookRepositoryImpl.java]: message should be more informative 
```suggestion
            throw new DataProcessingException("Can't save data", e);
```

File: src/main/java/com/example/demo/repository/BookRepositoryImpl.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/repository/BookRepositoryImpl.java:57]: add catch block

File: src/main/java/com/example/demo/repository/BookRepositoryImpl.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/repository/BookRepositoryImpl.java]:
```suggestion
            return query.getResultList();
```

File: src/main/java/com/example/demo/service/BookServiceImpl.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/service/BookServiceImpl.java:13]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/com/example/demo/service/BookServiceImpl.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/service/BookServiceImpl.java]:
```suggestion
    private final BookRepository bookRepository;
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's check task description
Add class with @SpringBootApplication annotation that will have a CommandLineRunner bean where you call BookService methods

Check IMPORTANT section
provide pom file
provide application.properties files

**Comment by layzimayner** [comment]
I created pom, application.properties and test/application.properties in last PR

File: pom.xml
Comment (Elena-Bruyako) [inline pom.xml]: remove dependency

File: src/main/java/com/example/demo/model/Book.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/model/Book.java:13]:
```suggestion
@Entity
@Getter
@Setter
```

File: src/main/java/com/example/demo/model/Book.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/model/Book.java]: remove all set and get methods

File: src/main/java/com/example/demo/model/Book.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/model/Book.java]: use for all required columns

```suggestion
    @Column(nullable = false)
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/example/demo/model/Book.java
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/model/Book.java]:
```suggestion
    @Column(nullable = false)
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
one comment

**Review by Elena-Bruyako — APPROVED** [review]

**Comment by Elena-Bruyako** [comment]
Changes were already approved, after approving merge PR, create branch and implement next HW
Screenshot 2024-11-15 at 11 45 54

**Pull request closed without merging**
